mcs/class/corlib/System.Runtime.Remoting.Proxies/RealProxy.cs

```diff
@@ -49,8 +49,45 @@ namespace System.Runtime.Remoting.Proxies
 	[StructLayout (LayoutKind.Sequential)]
 	internal class TransparentProxy {
 		public RealProxy _rp;
-		IntPtr _class;
+		unsafe Mono.RuntimeStructs.RemoteClass* _class;
 		bool _custom_type_info;
+
+		unsafe internal RuntimeType proxy_type() {
+			RuntimeTypeHandle h = Mono.RuntimeMarshaling.RuntimeTypeHandleFromClass (_class->proxy_class);
+			return (RuntimeType)Type.GetTypeFromHandle (h);
+		}
+
+		bool IsContextBoundObject {
+			get { return proxy_type ().IsContextful; }
+		}
+
+		Context TargetContext {
+			get { return _rp._targetContext; }
+		}
+
+		internal object load_remote_field_new (RuntimeTypeHandle type_handle, RuntimeFieldHandle field_handle) {
+
+			FieldInfo field = FieldInfo.GetFieldFromHandle (field_handle);
+			bool inCurrCtx =
+				IsContextBoundObject && TargetContext == Thread.CurrentContext;
+			if (inCurrCtx) {
+				object o = _rp._server;
+				return field.GetValue(o);
+			}
+
+			object[] in_args = new object[] { Type.GetTypeFromHandle(type_handle).FullName,
+							  field.Name };
+			MethodInfo minfo = typeof(object).GetMethod("FieldGetter", BindingFlags.NonPublic | BindingFlags.Instance);
+			if (minfo == null)
+				throw new ExecutionEngineException ("Couldn't find FieldGetter in object when making a method message");
+			MonoMethodMessage msg = new MonoMethodMessage (minfo, in_args, null);
```

Comment:
Why aren't you creating a size one object[] array and passing to MonoMethodMessage?